File: Lab6/src/main/java/com/example/lab6/data/Repository.java
package com.example.lab6.data;

import java.util.List;

public interface Repository {
    List<Apartment> getAll();
    Apartment getById(int id);
    List<Apartment> getAllAvailable(boolean bool);
    boolean addApartment(Apartment apartment);
    boolean deleteApartment(int id);
}

File: Lab6kotlin/src/main/kotlin/com/example/lab6/data/Repository.kt
package com.example.lab6.data

interface Repository {
    fun getAll(): List<Apartment>
    fun getById(id: Int): Apartment?
    fun getAllAvailable(bool: Boolean): List<Apartment>
    fun addApartment(apartment: Apartment): Boolean
    fun deleteApartment(id: Int): Boolean
}
